docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -48,10 +93,44 @@ For data sources that don't have partitioned data, a good strategy
 here is to simply return a single split for the entire table. This
 is the strategy employed by the Example HTTP connector.
 
+.. _connector-record-set-provider:
+
 ConnectorRecordSetProvider
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Given a split and a list of columns, the record set provider is
 responsible for delivering data to the Trino execution engine.
 It creates a ``RecordSet``, which in turn creates a ``RecordCursor``
 that is used by Trino to read the column values for each row.
+
+.. _connector-page-source-provider:
+
+ConnectorPageSourceProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Given a split and a list of columns, the page source provider is
+responsible for delivering data to the Trino execution engine.
+It creates a ``ConnectorPageSource``, which in turn creates ``Page`` objects
+that is used by Trino to read the column values.
+
+If not implemented, a default ``RecordPageSourceProvider`` will be used.
+Given a record set provider, it returns an instance of ``RecordPageSource``
+that builds ``Page`` objects from records in a record set.
+
+A connector can implement a page source provider instead of the record set
+provider when it's better to create pages directly, instead of individual
+records that would be converted to pages later anyway.
+
+To add support for updating and/or deleting rows in a connector, it needs
+to implement a ``ConnectorPageSourceProvider`` that returns
+an ``UpdatablePageSource``.
```

Comment:
🤦 I just realized there's a whole separate page for delete and update support. I'll add a link here.